### PR TITLE
Feature/custom mask

### PR DESCRIPTION
**Pull request by lastMove**
#464 

- Add Custom to MaskType 
- Add a demo case in the playground App 
- Introduce `PathProvider` typeAlias
`public typealias CustomMaskProvider = (CGSize) -> UIBezierPath`

### CODE REVIEW

**Comment by lastMove** [comment]
The PR is now ready, could someone review it please ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ N/A
 
 #### Enhancements
 
+- Introduce `Custom` MaskType, you can now use `MaskDesignable` with custom masks. [#465](https://github.com/IBAnimatable/IBAnimatable/pull/465) by [@lastmove](https://github.com/lastmove)
```
Comment (tbaranes) [inline CHANGELOG.md:15]: Lowercase

File: Documentation/APIs.md [hunk 2]
```diff
@@ -95,6 +95,7 @@ To use `IBAnimatable`, we can drag and drop a UIKit element and connect it with
 * `Star`: Can also specify the points of the Star, e.g. use `Star(6)` to have a star with 6 points. If not specified, the default is 5 points. 
 * `Wave`: Can use parameters to customize the `Wave`. `Wave(down, 20, 5)` means the Wave faces down, width is 20 and offset is 5. If not specified, the default value is `Wave(up, 40, 0)`. 
 * `Parallelogram`: Can use parameters to customize the `Parallelogram`. `Parallelogram(60)` means the top left angle of the Parallelogram will have an angle of 60 degrees. The default value is `Parallelogram(60)`. if angle == 90 it will be a rectangular Mask. if angle < 90 the parallelogram will be oriented to left. 
+* `Custom`: Allows you to use your own bezier path as mask. Only usable from code (not from IB). You have to pass in parameter a closure that takes a `CGSize` (the current's view Size) and returns the `UIBezierPath`
```
Comment (tbaranes) [inline Documentation/APIs.md]: Why is it a bug fix?

File: IBAnimatableApp/MaskViewController.swift [hunk 3]
```diff
@@ -76,7 +77,74 @@ extension MaskViewController : UIPickerViewDelegate, UIPickerViewDataSource {
...
+    bezierPath.addCurve(to: CGPoint(x: frame.minX + 28.79, y: frame.minY + 13.5),
+controlPoint1: CGPoint(x: frame.minX + 22.18, y: frame.minY + 13.5),
+controlPoint2: CGPoint(x: frame.minX + 24.38, y: frame.minY + 13.5))
+    bezierPath.addLine(to: CGPoint(x: frame.maxX - 16.79, y: frame.minY + 13.5))
+    bezierPath.addCurve(to: CGPoint(x: frame.maxX - 8.2, y: frame.minY + 14.15),
+controlPoint1: CGPoint(x: frame.maxX - 12.38, y: frame.minY + 13.5),
+controlPoint2: CGPoint(x: frame.maxX - 10.18, y: frame.minY + 13.5))
+    bezierPath.close()
+    return bezierPath
+
```
Comment (tbaranes) [inline IBAnimatableApp/MaskViewController.swift]: Removing newline ?

**Comment by tbaranes** [comment]
Approved 👍
(Can't do it on this app)

**Comment by lastMove** [comment]
Thanks ;)

**Comment by JakeLin** [comment]
@lastMove great feature 👍👍👍 If you want to see you commits in https://github.com/IBAnimatable/IBAnimatable/graphs/contributors, please associate your email or use the GitHub login email to commit.

**Comment by lastMove** [comment]
Oh, I didn't know about this, Thanks !

**Comment by DanielAsher** [comment]
Looking forward to playing with this!

**Pull request merged**